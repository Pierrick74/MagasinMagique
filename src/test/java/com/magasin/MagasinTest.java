package com.magasin;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

class MagasinTest {

    @Test
    public void testList() {
        String[] names = new String[] {"Comté", "Kryptonite", "Pass VIP Concert", "kryptonite", "comté"};
        Integer[] sellins = new Integer[] {8, -1, 0, 2, 4, 5, 6, 9, 10, 11};
        Integer[] qualities = new Integer[] {1, 0, 20, 50, 49, 51, 80, };

        CombinationApprovals.verifyAllCombinations(
                (String name, Integer sellin, Integer quality) -> {
                    Item[] items = new Item[] { new Item(name, sellin, quality) };
                    Magasin app = new Magasin(items);
                    app.updateQuality();
                    return items[0].toString();
                },
                names, sellins, qualities
        );
    }

    @Nested
    class MagicPowerTest {

        @ParameterizedTest(name = "Item {0}: sellIn {1}→{3}, quality {2}→{4}")
        @DisplayName("given name {0}, selin at {1} and quality at {2} When Update then selin = {3} and quality = {4}")
        @CsvSource(value = {
                "'Pouvoirs magiques', 5, 11, 4, 9",
                "'Pouvoirs magiques', 1, 11, 0, 9",
                "'Pouvoirs magiques', 0, 11, -1, 7",
                "'Pouvoirs magiques', 5, 0, 4, 0"
        })
        public void testMagicPower(String name, Integer sellin, Integer quality, Integer resultSellin, Integer resultQuality ) {
            Item[] items = new Item[] { new Item(name, sellin, quality) };
            Magasin app = new Magasin(items);
            app.updateQuality();
            Assertions.assertEquals(items[0].quality, resultQuality);
            Assertions.assertEquals(items[0].sellIn, resultSellin);
        }
    }
}
