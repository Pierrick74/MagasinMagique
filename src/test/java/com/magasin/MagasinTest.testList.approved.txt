[Comté, 8, 1] => Comté, 7, 2 
[Comté, 8, 0] => Comté, 7, 1 
[Comté, 8, 20] => Comté, 7, 21 
[Comté, 8, 50] => Comté, 7, 50 
[Comté, 8, 49] => Comté, 7, 50 
[Comté, 8, 51] => Comté, 7, 51 
[Comté, 8, 80] => Comté, 7, 80 
[Comté, -1, 1] => Comté, -2, 3 
[Comté, -1, 0] => Comté, -2, 2 
[Comté, -1, 20] => Comté, -2, 22 
[Comté, -1, 50] => Comté, -2, 50 
[Comté, -1, 49] => Comté, -2, 50 
[Comté, -1, 51] => Comté, -2, 51 
[Comté, -1, 80] => Comté, -2, 80 
[Comté, 0, 1] => Comté, -1, 3 
[Comté, 0, 0] => Comté, -1, 2 
[Comté, 0, 20] => Comté, -1, 22 
[Comté, 0, 50] => Comté, -1, 50 
[Comté, 0, 49] => Comté, -1, 50 
[Comté, 0, 51] => Comté, -1, 51 
[Comté, 0, 80] => Comté, -1, 80 
[Comté, 2, 1] => Comté, 1, 2 
[Comté, 2, 0] => Comté, 1, 1 
[Comté, 2, 20] => Comté, 1, 21 
[Comté, 2, 50] => Comté, 1, 50 
[Comté, 2, 49] => Comté, 1, 50 
[Comté, 2, 51] => Comté, 1, 51 
[Comté, 2, 80] => Comté, 1, 80 
[Comté, 4, 1] => Comté, 3, 2 
[Comté, 4, 0] => Comté, 3, 1 
[Comté, 4, 20] => Comté, 3, 21 
[Comté, 4, 50] => Comté, 3, 50 
[Comté, 4, 49] => Comté, 3, 50 
[Comté, 4, 51] => Comté, 3, 51 
[Comté, 4, 80] => Comté, 3, 80 
[Comté, 5, 1] => Comté, 4, 2 
[Comté, 5, 0] => Comté, 4, 1 
[Comté, 5, 20] => Comté, 4, 21 
[Comté, 5, 50] => Comté, 4, 50 
[Comté, 5, 49] => Comté, 4, 50 
[Comté, 5, 51] => Comté, 4, 51 
[Comté, 5, 80] => Comté, 4, 80 
[Comté, 6, 1] => Comté, 5, 2 
[Comté, 6, 0] => Comté, 5, 1 
[Comté, 6, 20] => Comté, 5, 21 
[Comté, 6, 50] => Comté, 5, 50 
[Comté, 6, 49] => Comté, 5, 50 
[Comté, 6, 51] => Comté, 5, 51 
[Comté, 6, 80] => Comté, 5, 80 
[Comté, 9, 1] => Comté, 8, 2 
[Comté, 9, 0] => Comté, 8, 1 
[Comté, 9, 20] => Comté, 8, 21 
[Comté, 9, 50] => Comté, 8, 50 
[Comté, 9, 49] => Comté, 8, 50 
[Comté, 9, 51] => Comté, 8, 51 
[Comté, 9, 80] => Comté, 8, 80 
[Comté, 10, 1] => Comté, 9, 2 
[Comté, 10, 0] => Comté, 9, 1 
[Comté, 10, 20] => Comté, 9, 21 
[Comté, 10, 50] => Comté, 9, 50 
[Comté, 10, 49] => Comté, 9, 50 
[Comté, 10, 51] => Comté, 9, 51 
[Comté, 10, 80] => Comté, 9, 80 
[Comté, 11, 1] => Comté, 10, 2 
[Comté, 11, 0] => Comté, 10, 1 
[Comté, 11, 20] => Comté, 10, 21 
[Comté, 11, 50] => Comté, 10, 50 
[Comté, 11, 49] => Comté, 10, 50 
[Comté, 11, 51] => Comté, 10, 51 
[Comté, 11, 80] => Comté, 10, 80 
[Kryptonite, 8, 1] => Kryptonite, 8, 1 
[Kryptonite, 8, 0] => Kryptonite, 8, 0 
[Kryptonite, 8, 20] => Kryptonite, 8, 20 
[Kryptonite, 8, 50] => Kryptonite, 8, 50 
[Kryptonite, 8, 49] => Kryptonite, 8, 49 
[Kryptonite, 8, 51] => Kryptonite, 8, 51 
[Kryptonite, 8, 80] => Kryptonite, 8, 80 
[Kryptonite, -1, 1] => Kryptonite, -1, 1 
[Kryptonite, -1, 0] => Kryptonite, -1, 0 
[Kryptonite, -1, 20] => Kryptonite, -1, 20 
[Kryptonite, -1, 50] => Kryptonite, -1, 50 
[Kryptonite, -1, 49] => Kryptonite, -1, 49 
[Kryptonite, -1, 51] => Kryptonite, -1, 51 
[Kryptonite, -1, 80] => Kryptonite, -1, 80 
[Kryptonite, 0, 1] => Kryptonite, 0, 1 
[Kryptonite, 0, 0] => Kryptonite, 0, 0 
[Kryptonite, 0, 20] => Kryptonite, 0, 20 
[Kryptonite, 0, 50] => Kryptonite, 0, 50 
[Kryptonite, 0, 49] => Kryptonite, 0, 49 
[Kryptonite, 0, 51] => Kryptonite, 0, 51 
[Kryptonite, 0, 80] => Kryptonite, 0, 80 
[Kryptonite, 2, 1] => Kryptonite, 2, 1 
[Kryptonite, 2, 0] => Kryptonite, 2, 0 
[Kryptonite, 2, 20] => Kryptonite, 2, 20 
[Kryptonite, 2, 50] => Kryptonite, 2, 50 
[Kryptonite, 2, 49] => Kryptonite, 2, 49 
[Kryptonite, 2, 51] => Kryptonite, 2, 51 
[Kryptonite, 2, 80] => Kryptonite, 2, 80 
[Kryptonite, 4, 1] => Kryptonite, 4, 1 
[Kryptonite, 4, 0] => Kryptonite, 4, 0 
[Kryptonite, 4, 20] => Kryptonite, 4, 20 
[Kryptonite, 4, 50] => Kryptonite, 4, 50 
[Kryptonite, 4, 49] => Kryptonite, 4, 49 
[Kryptonite, 4, 51] => Kryptonite, 4, 51 
[Kryptonite, 4, 80] => Kryptonite, 4, 80 
[Kryptonite, 5, 1] => Kryptonite, 5, 1 
[Kryptonite, 5, 0] => Kryptonite, 5, 0 
[Kryptonite, 5, 20] => Kryptonite, 5, 20 
[Kryptonite, 5, 50] => Kryptonite, 5, 50 
[Kryptonite, 5, 49] => Kryptonite, 5, 49 
[Kryptonite, 5, 51] => Kryptonite, 5, 51 
[Kryptonite, 5, 80] => Kryptonite, 5, 80 
[Kryptonite, 6, 1] => Kryptonite, 6, 1 
[Kryptonite, 6, 0] => Kryptonite, 6, 0 
[Kryptonite, 6, 20] => Kryptonite, 6, 20 
[Kryptonite, 6, 50] => Kryptonite, 6, 50 
[Kryptonite, 6, 49] => Kryptonite, 6, 49 
[Kryptonite, 6, 51] => Kryptonite, 6, 51 
[Kryptonite, 6, 80] => Kryptonite, 6, 80 
[Kryptonite, 9, 1] => Kryptonite, 9, 1 
[Kryptonite, 9, 0] => Kryptonite, 9, 0 
[Kryptonite, 9, 20] => Kryptonite, 9, 20 
[Kryptonite, 9, 50] => Kryptonite, 9, 50 
[Kryptonite, 9, 49] => Kryptonite, 9, 49 
[Kryptonite, 9, 51] => Kryptonite, 9, 51 
[Kryptonite, 9, 80] => Kryptonite, 9, 80 
[Kryptonite, 10, 1] => Kryptonite, 10, 1 
[Kryptonite, 10, 0] => Kryptonite, 10, 0 
[Kryptonite, 10, 20] => Kryptonite, 10, 20 
[Kryptonite, 10, 50] => Kryptonite, 10, 50 
[Kryptonite, 10, 49] => Kryptonite, 10, 49 
[Kryptonite, 10, 51] => Kryptonite, 10, 51 
[Kryptonite, 10, 80] => Kryptonite, 10, 80 
[Kryptonite, 11, 1] => Kryptonite, 11, 1 
[Kryptonite, 11, 0] => Kryptonite, 11, 0 
[Kryptonite, 11, 20] => Kryptonite, 11, 20 
[Kryptonite, 11, 50] => Kryptonite, 11, 50 
[Kryptonite, 11, 49] => Kryptonite, 11, 49 
[Kryptonite, 11, 51] => Kryptonite, 11, 51 
[Kryptonite, 11, 80] => Kryptonite, 11, 80 
[Pass VIP Concert, 8, 1] => Pass VIP Concert, 7, 3 
[Pass VIP Concert, 8, 0] => Pass VIP Concert, 7, 2 
[Pass VIP Concert, 8, 20] => Pass VIP Concert, 7, 22 
[Pass VIP Concert, 8, 50] => Pass VIP Concert, 7, 50 
[Pass VIP Concert, 8, 49] => Pass VIP Concert, 7, 50 
[Pass VIP Concert, 8, 51] => Pass VIP Concert, 7, 51 
[Pass VIP Concert, 8, 80] => Pass VIP Concert, 7, 80 
[Pass VIP Concert, -1, 1] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, -1, 0] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, -1, 20] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, -1, 50] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, -1, 49] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, -1, 51] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, -1, 80] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, 0, 1] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 0, 0] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 0, 20] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 0, 50] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 0, 49] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 0, 51] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 0, 80] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 2, 1] => Pass VIP Concert, 1, 4 
[Pass VIP Concert, 2, 0] => Pass VIP Concert, 1, 3 
[Pass VIP Concert, 2, 20] => Pass VIP Concert, 1, 23 
[Pass VIP Concert, 2, 50] => Pass VIP Concert, 1, 50 
[Pass VIP Concert, 2, 49] => Pass VIP Concert, 1, 50 
[Pass VIP Concert, 2, 51] => Pass VIP Concert, 1, 51 
[Pass VIP Concert, 2, 80] => Pass VIP Concert, 1, 80 
[Pass VIP Concert, 4, 1] => Pass VIP Concert, 3, 4 
[Pass VIP Concert, 4, 0] => Pass VIP Concert, 3, 3 
[Pass VIP Concert, 4, 20] => Pass VIP Concert, 3, 23 
[Pass VIP Concert, 4, 50] => Pass VIP Concert, 3, 50 
[Pass VIP Concert, 4, 49] => Pass VIP Concert, 3, 50 
[Pass VIP Concert, 4, 51] => Pass VIP Concert, 3, 51 
[Pass VIP Concert, 4, 80] => Pass VIP Concert, 3, 80 
[Pass VIP Concert, 5, 1] => Pass VIP Concert, 4, 4 
[Pass VIP Concert, 5, 0] => Pass VIP Concert, 4, 3 
[Pass VIP Concert, 5, 20] => Pass VIP Concert, 4, 23 
[Pass VIP Concert, 5, 50] => Pass VIP Concert, 4, 50 
[Pass VIP Concert, 5, 49] => Pass VIP Concert, 4, 50 
[Pass VIP Concert, 5, 51] => Pass VIP Concert, 4, 51 
[Pass VIP Concert, 5, 80] => Pass VIP Concert, 4, 80 
[Pass VIP Concert, 6, 1] => Pass VIP Concert, 5, 3 
[Pass VIP Concert, 6, 0] => Pass VIP Concert, 5, 2 
[Pass VIP Concert, 6, 20] => Pass VIP Concert, 5, 22 
[Pass VIP Concert, 6, 50] => Pass VIP Concert, 5, 50 
[Pass VIP Concert, 6, 49] => Pass VIP Concert, 5, 50 
[Pass VIP Concert, 6, 51] => Pass VIP Concert, 5, 51 
[Pass VIP Concert, 6, 80] => Pass VIP Concert, 5, 80 
[Pass VIP Concert, 9, 1] => Pass VIP Concert, 8, 3 
[Pass VIP Concert, 9, 0] => Pass VIP Concert, 8, 2 
[Pass VIP Concert, 9, 20] => Pass VIP Concert, 8, 22 
[Pass VIP Concert, 9, 50] => Pass VIP Concert, 8, 50 
[Pass VIP Concert, 9, 49] => Pass VIP Concert, 8, 50 
[Pass VIP Concert, 9, 51] => Pass VIP Concert, 8, 51 
[Pass VIP Concert, 9, 80] => Pass VIP Concert, 8, 80 
[Pass VIP Concert, 10, 1] => Pass VIP Concert, 9, 3 
[Pass VIP Concert, 10, 0] => Pass VIP Concert, 9, 2 
[Pass VIP Concert, 10, 20] => Pass VIP Concert, 9, 22 
[Pass VIP Concert, 10, 50] => Pass VIP Concert, 9, 50 
[Pass VIP Concert, 10, 49] => Pass VIP Concert, 9, 50 
[Pass VIP Concert, 10, 51] => Pass VIP Concert, 9, 51 
[Pass VIP Concert, 10, 80] => Pass VIP Concert, 9, 80 
[Pass VIP Concert, 11, 1] => Pass VIP Concert, 10, 2 
[Pass VIP Concert, 11, 0] => Pass VIP Concert, 10, 1 
[Pass VIP Concert, 11, 20] => Pass VIP Concert, 10, 21 
[Pass VIP Concert, 11, 50] => Pass VIP Concert, 10, 50 
[Pass VIP Concert, 11, 49] => Pass VIP Concert, 10, 50 
[Pass VIP Concert, 11, 51] => Pass VIP Concert, 10, 51 
[Pass VIP Concert, 11, 80] => Pass VIP Concert, 10, 80 
[Pouvoirs magiques, 8, 1] => Pouvoirs magiques, 7, 0 
[Pouvoirs magiques, 8, 0] => Pouvoirs magiques, 7, 0 
[Pouvoirs magiques, 8, 20] => Pouvoirs magiques, 7, 19 
[Pouvoirs magiques, 8, 50] => Pouvoirs magiques, 7, 49 
[Pouvoirs magiques, 8, 49] => Pouvoirs magiques, 7, 48 
[Pouvoirs magiques, 8, 51] => Pouvoirs magiques, 7, 50 
[Pouvoirs magiques, 8, 80] => Pouvoirs magiques, 7, 79 
[Pouvoirs magiques, -1, 1] => Pouvoirs magiques, -2, 0 
[Pouvoirs magiques, -1, 0] => Pouvoirs magiques, -2, 0 
[Pouvoirs magiques, -1, 20] => Pouvoirs magiques, -2, 18 
[Pouvoirs magiques, -1, 50] => Pouvoirs magiques, -2, 48 
[Pouvoirs magiques, -1, 49] => Pouvoirs magiques, -2, 47 
[Pouvoirs magiques, -1, 51] => Pouvoirs magiques, -2, 49 
[Pouvoirs magiques, -1, 80] => Pouvoirs magiques, -2, 78 
[Pouvoirs magiques, 0, 1] => Pouvoirs magiques, -1, 0 
[Pouvoirs magiques, 0, 0] => Pouvoirs magiques, -1, 0 
[Pouvoirs magiques, 0, 20] => Pouvoirs magiques, -1, 18 
[Pouvoirs magiques, 0, 50] => Pouvoirs magiques, -1, 48 
[Pouvoirs magiques, 0, 49] => Pouvoirs magiques, -1, 47 
[Pouvoirs magiques, 0, 51] => Pouvoirs magiques, -1, 49 
[Pouvoirs magiques, 0, 80] => Pouvoirs magiques, -1, 78 
[Pouvoirs magiques, 2, 1] => Pouvoirs magiques, 1, 0 
[Pouvoirs magiques, 2, 0] => Pouvoirs magiques, 1, 0 
[Pouvoirs magiques, 2, 20] => Pouvoirs magiques, 1, 19 
[Pouvoirs magiques, 2, 50] => Pouvoirs magiques, 1, 49 
[Pouvoirs magiques, 2, 49] => Pouvoirs magiques, 1, 48 
[Pouvoirs magiques, 2, 51] => Pouvoirs magiques, 1, 50 
[Pouvoirs magiques, 2, 80] => Pouvoirs magiques, 1, 79 
[Pouvoirs magiques, 4, 1] => Pouvoirs magiques, 3, 0 
[Pouvoirs magiques, 4, 0] => Pouvoirs magiques, 3, 0 
[Pouvoirs magiques, 4, 20] => Pouvoirs magiques, 3, 19 
[Pouvoirs magiques, 4, 50] => Pouvoirs magiques, 3, 49 
[Pouvoirs magiques, 4, 49] => Pouvoirs magiques, 3, 48 
[Pouvoirs magiques, 4, 51] => Pouvoirs magiques, 3, 50 
[Pouvoirs magiques, 4, 80] => Pouvoirs magiques, 3, 79 
[Pouvoirs magiques, 5, 1] => Pouvoirs magiques, 4, 0 
[Pouvoirs magiques, 5, 0] => Pouvoirs magiques, 4, 0 
[Pouvoirs magiques, 5, 20] => Pouvoirs magiques, 4, 19 
[Pouvoirs magiques, 5, 50] => Pouvoirs magiques, 4, 49 
[Pouvoirs magiques, 5, 49] => Pouvoirs magiques, 4, 48 
[Pouvoirs magiques, 5, 51] => Pouvoirs magiques, 4, 50 
[Pouvoirs magiques, 5, 80] => Pouvoirs magiques, 4, 79 
[Pouvoirs magiques, 6, 1] => Pouvoirs magiques, 5, 0 
[Pouvoirs magiques, 6, 0] => Pouvoirs magiques, 5, 0 
[Pouvoirs magiques, 6, 20] => Pouvoirs magiques, 5, 19 
[Pouvoirs magiques, 6, 50] => Pouvoirs magiques, 5, 49 
[Pouvoirs magiques, 6, 49] => Pouvoirs magiques, 5, 48 
[Pouvoirs magiques, 6, 51] => Pouvoirs magiques, 5, 50 
[Pouvoirs magiques, 6, 80] => Pouvoirs magiques, 5, 79 
[Pouvoirs magiques, 9, 1] => Pouvoirs magiques, 8, 0 
[Pouvoirs magiques, 9, 0] => Pouvoirs magiques, 8, 0 
[Pouvoirs magiques, 9, 20] => Pouvoirs magiques, 8, 19 
[Pouvoirs magiques, 9, 50] => Pouvoirs magiques, 8, 49 
[Pouvoirs magiques, 9, 49] => Pouvoirs magiques, 8, 48 
[Pouvoirs magiques, 9, 51] => Pouvoirs magiques, 8, 50 
[Pouvoirs magiques, 9, 80] => Pouvoirs magiques, 8, 79 
[Pouvoirs magiques, 10, 1] => Pouvoirs magiques, 9, 0 
[Pouvoirs magiques, 10, 0] => Pouvoirs magiques, 9, 0 
[Pouvoirs magiques, 10, 20] => Pouvoirs magiques, 9, 19 
[Pouvoirs magiques, 10, 50] => Pouvoirs magiques, 9, 49 
[Pouvoirs magiques, 10, 49] => Pouvoirs magiques, 9, 48 
[Pouvoirs magiques, 10, 51] => Pouvoirs magiques, 9, 50 
[Pouvoirs magiques, 10, 80] => Pouvoirs magiques, 9, 79 
[Pouvoirs magiques, 11, 1] => Pouvoirs magiques, 10, 0 
[Pouvoirs magiques, 11, 0] => Pouvoirs magiques, 10, 0 
[Pouvoirs magiques, 11, 20] => Pouvoirs magiques, 10, 19 
[Pouvoirs magiques, 11, 50] => Pouvoirs magiques, 10, 49 
[Pouvoirs magiques, 11, 49] => Pouvoirs magiques, 10, 48 
[Pouvoirs magiques, 11, 51] => Pouvoirs magiques, 10, 50 
[Pouvoirs magiques, 11, 80] => Pouvoirs magiques, 10, 79 
[kryptonite, 8, 1] => kryptonite, 7, 0 
[kryptonite, 8, 0] => kryptonite, 7, 0 
[kryptonite, 8, 20] => kryptonite, 7, 19 
[kryptonite, 8, 50] => kryptonite, 7, 49 
[kryptonite, 8, 49] => kryptonite, 7, 48 
[kryptonite, 8, 51] => kryptonite, 7, 50 
[kryptonite, 8, 80] => kryptonite, 7, 79 
[kryptonite, -1, 1] => kryptonite, -2, 0 
[kryptonite, -1, 0] => kryptonite, -2, 0 
[kryptonite, -1, 20] => kryptonite, -2, 18 
[kryptonite, -1, 50] => kryptonite, -2, 48 
[kryptonite, -1, 49] => kryptonite, -2, 47 
[kryptonite, -1, 51] => kryptonite, -2, 49 
[kryptonite, -1, 80] => kryptonite, -2, 78 
[kryptonite, 0, 1] => kryptonite, -1, 0 
[kryptonite, 0, 0] => kryptonite, -1, 0 
[kryptonite, 0, 20] => kryptonite, -1, 18 
[kryptonite, 0, 50] => kryptonite, -1, 48 
[kryptonite, 0, 49] => kryptonite, -1, 47 
[kryptonite, 0, 51] => kryptonite, -1, 49 
[kryptonite, 0, 80] => kryptonite, -1, 78 
[kryptonite, 2, 1] => kryptonite, 1, 0 
[kryptonite, 2, 0] => kryptonite, 1, 0 
[kryptonite, 2, 20] => kryptonite, 1, 19 
[kryptonite, 2, 50] => kryptonite, 1, 49 
[kryptonite, 2, 49] => kryptonite, 1, 48 
[kryptonite, 2, 51] => kryptonite, 1, 50 
[kryptonite, 2, 80] => kryptonite, 1, 79 
[kryptonite, 4, 1] => kryptonite, 3, 0 
[kryptonite, 4, 0] => kryptonite, 3, 0 
[kryptonite, 4, 20] => kryptonite, 3, 19 
[kryptonite, 4, 50] => kryptonite, 3, 49 
[kryptonite, 4, 49] => kryptonite, 3, 48 
[kryptonite, 4, 51] => kryptonite, 3, 50 
[kryptonite, 4, 80] => kryptonite, 3, 79 
[kryptonite, 5, 1] => kryptonite, 4, 0 
[kryptonite, 5, 0] => kryptonite, 4, 0 
[kryptonite, 5, 20] => kryptonite, 4, 19 
[kryptonite, 5, 50] => kryptonite, 4, 49 
[kryptonite, 5, 49] => kryptonite, 4, 48 
[kryptonite, 5, 51] => kryptonite, 4, 50 
[kryptonite, 5, 80] => kryptonite, 4, 79 
[kryptonite, 6, 1] => kryptonite, 5, 0 
[kryptonite, 6, 0] => kryptonite, 5, 0 
[kryptonite, 6, 20] => kryptonite, 5, 19 
[kryptonite, 6, 50] => kryptonite, 5, 49 
[kryptonite, 6, 49] => kryptonite, 5, 48 
[kryptonite, 6, 51] => kryptonite, 5, 50 
[kryptonite, 6, 80] => kryptonite, 5, 79 
[kryptonite, 9, 1] => kryptonite, 8, 0 
[kryptonite, 9, 0] => kryptonite, 8, 0 
[kryptonite, 9, 20] => kryptonite, 8, 19 
[kryptonite, 9, 50] => kryptonite, 8, 49 
[kryptonite, 9, 49] => kryptonite, 8, 48 
[kryptonite, 9, 51] => kryptonite, 8, 50 
[kryptonite, 9, 80] => kryptonite, 8, 79 
[kryptonite, 10, 1] => kryptonite, 9, 0 
[kryptonite, 10, 0] => kryptonite, 9, 0 
[kryptonite, 10, 20] => kryptonite, 9, 19 
[kryptonite, 10, 50] => kryptonite, 9, 49 
[kryptonite, 10, 49] => kryptonite, 9, 48 
[kryptonite, 10, 51] => kryptonite, 9, 50 
[kryptonite, 10, 80] => kryptonite, 9, 79 
[kryptonite, 11, 1] => kryptonite, 10, 0 
[kryptonite, 11, 0] => kryptonite, 10, 0 
[kryptonite, 11, 20] => kryptonite, 10, 19 
[kryptonite, 11, 50] => kryptonite, 10, 49 
[kryptonite, 11, 49] => kryptonite, 10, 48 
[kryptonite, 11, 51] => kryptonite, 10, 50 
[kryptonite, 11, 80] => kryptonite, 10, 79 
[comté, 8, 1] => comté, 7, 0 
[comté, 8, 0] => comté, 7, 0 
[comté, 8, 20] => comté, 7, 19 
[comté, 8, 50] => comté, 7, 49 
[comté, 8, 49] => comté, 7, 48 
[comté, 8, 51] => comté, 7, 50 
[comté, 8, 80] => comté, 7, 79 
[comté, -1, 1] => comté, -2, 0 
[comté, -1, 0] => comté, -2, 0 
[comté, -1, 20] => comté, -2, 18 
[comté, -1, 50] => comté, -2, 48 
[comté, -1, 49] => comté, -2, 47 
[comté, -1, 51] => comté, -2, 49 
[comté, -1, 80] => comté, -2, 78 
[comté, 0, 1] => comté, -1, 0 
[comté, 0, 0] => comté, -1, 0 
[comté, 0, 20] => comté, -1, 18 
[comté, 0, 50] => comté, -1, 48 
[comté, 0, 49] => comté, -1, 47 
[comté, 0, 51] => comté, -1, 49 
[comté, 0, 80] => comté, -1, 78 
[comté, 2, 1] => comté, 1, 0 
[comté, 2, 0] => comté, 1, 0 
[comté, 2, 20] => comté, 1, 19 
[comté, 2, 50] => comté, 1, 49 
[comté, 2, 49] => comté, 1, 48 
[comté, 2, 51] => comté, 1, 50 
[comté, 2, 80] => comté, 1, 79 
[comté, 4, 1] => comté, 3, 0 
[comté, 4, 0] => comté, 3, 0 
[comté, 4, 20] => comté, 3, 19 
[comté, 4, 50] => comté, 3, 49 
[comté, 4, 49] => comté, 3, 48 
[comté, 4, 51] => comté, 3, 50 
[comté, 4, 80] => comté, 3, 79 
[comté, 5, 1] => comté, 4, 0 
[comté, 5, 0] => comté, 4, 0 
[comté, 5, 20] => comté, 4, 19 
[comté, 5, 50] => comté, 4, 49 
[comté, 5, 49] => comté, 4, 48 
[comté, 5, 51] => comté, 4, 50 
[comté, 5, 80] => comté, 4, 79 
[comté, 6, 1] => comté, 5, 0 
[comté, 6, 0] => comté, 5, 0 
[comté, 6, 20] => comté, 5, 19 
[comté, 6, 50] => comté, 5, 49 
[comté, 6, 49] => comté, 5, 48 
[comté, 6, 51] => comté, 5, 50 
[comté, 6, 80] => comté, 5, 79 
[comté, 9, 1] => comté, 8, 0 
[comté, 9, 0] => comté, 8, 0 
[comté, 9, 20] => comté, 8, 19 
[comté, 9, 50] => comté, 8, 49 
[comté, 9, 49] => comté, 8, 48 
[comté, 9, 51] => comté, 8, 50 
[comté, 9, 80] => comté, 8, 79 
[comté, 10, 1] => comté, 9, 0 
[comté, 10, 0] => comté, 9, 0 
[comté, 10, 20] => comté, 9, 19 
[comté, 10, 50] => comté, 9, 49 
[comté, 10, 49] => comté, 9, 48 
[comté, 10, 51] => comté, 9, 50 
[comté, 10, 80] => comté, 9, 79 
[comté, 11, 1] => comté, 10, 0 
[comté, 11, 0] => comté, 10, 0 
[comté, 11, 20] => comté, 10, 19 
[comté, 11, 50] => comté, 10, 49 
[comté, 11, 49] => comté, 10, 48 
[comté, 11, 51] => comté, 10, 50 
[comté, 11, 80] => comté, 10, 79 
